core/Updates/5.2.0-b2.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link    https://matomo.org
+ * @license https://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Updates;
+
+use Piwik\Common;
+use Piwik\Config;
+use Piwik\Db;
+use Piwik\Updater;
+use Piwik\Updater\Migration\Factory as MigrationFactory;
+use Piwik\Updates;
+
+class Updates_5_2_0_b2 extends Updates
+{
+    /**
+     * @var MigrationFactory
+     */
+    private $migration;
+
+    public function __construct(MigrationFactory $factory)
+    {
+        $this->migration = $factory;
+    }
+
+    public function getMigrations(Updater $updater)
+    {
+        $migrations = [];
+
+        $config = Config::getInstance();
+        $dbConfig = $config->database;
+
+        // only run migration if config is not set
+        if (empty($dbConfig['collation'])) {
+            try {
+                $db = Db::get();
+                $userTable = Common::prefixTable('user');
+                $userTableStatus = $db->fetchRow('SHOW TABLE STATUS WHERE Name = ?', [$userTable]);
+                $connectionCollation = $db->fetchOne('SELECT @@collation_connection');
+
+                // only update config if user table and connection report same collation
+                if (
```

Comment:
I guess this check is needed to ensure we only set the collation automatically, if the default collation hasn't changed since the user table was created?
I'm wondering if it could make sense to also look at the latest (archive) table that has been created and if it's collation matches the user table, we could safely use it, even if the connection collation is different?
This might make it possible to also cover cases where people might update Matomo right after updating MariaDb...

Reply:
Should be a good to look at the archive tables as a fallback if the connection is already problematic :+1: